panther_hardware_interfaces/include/panther_hardware_interfaces/roboteq_driver.hpp

```diff
@@ -0,0 +1,198 @@
+
+  bool IsCanError() const { return can_error_.load(); }
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadTemperature();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::uint16_t ReadVoltage();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadBatAmps1();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadBatAmps2();
+
+  /**
+   * @brief Reads all the PDO data returned from Roboteq (motors feedback, error flags) and saves
+   * current timestamp
+   */
+  RoboteqDriverFeedback ReadRoboteqDriverFeedback();
+
+  /**
+   * @param cmd command value in the range [-1000, 1000]
+   * @exception std::runtime_error if operation fails
+   */
+  void SendRoboteqCmdChannel1(const std::int32_t cmd);
+
+  /**
+   * @param cmd command value in the range [-1000, 1000]
+   * @exception std::runtime_error if operation fails
+   */
+  void SendRoboteqCmdChannel2(const std::int32_t cmd);
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void ResetRoboteqScript();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOnEstop();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOffEstop();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOnSafetyStopChannel1();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOnSafetyStopChannel2();
+
+private:
+  /**
+   * @brief Blocking SDO read operation
+   *
+   * @exception std::runtime_error if operation fails
+   */
+  template <typename T>
+  T SyncSdoRead(const std::uint16_t index, const std::uint8_t subindex);
+
+  /**
+   * @brief Blocking SDO write operation
+   *
+   * @exception std::runtime_error if operation fails
+   */
+  template <typename T>
+  void SyncSdoWrite(const std::uint16_t index, const std::uint8_t subindex, const T data);
+
+  void OnBoot(
+    const lely::canopen::NmtState st, const char es, const std::string & what) noexcept override;
+  void OnRpdoWrite(const std::uint16_t idx, const std::uint8_t subidx) noexcept override;
+  void OnCanError(const lely::io::CanError /* error */) noexcept override
+  {
+    can_error_.store(true);
+  }
+
+  // emcy - emergency - I don't think that it is used by Roboteq - haven't found any information
+  // about it while ros2_canopen has the ability to read it, I didn't see any attempts to handle it
+  // void OnEmcy(std::uint16_t eec, std::uint8_t er, std::uint8_t msef[5]) noexcept override;
+
+  std::atomic_bool booted_ = false;
+  std::condition_variable boot_cond_var_;
+  std::mutex boot_mtx_;
+  std::string boot_error_str_;
+
+  std::atomic_bool can_error_;
+
+  timespec last_rpdo_write_timestamp_;
+  std::mutex rpdo_timestamp_mtx_;
+
+  const std::chrono::milliseconds sdo_operation_timeout_;
```

Comment:
```suggestion
  const std::chrono::milliseconds sdo_operation_timeout_ms_;
```
Personal preference, if time is not in seconds